packages/postgres/test/unit/postgres_p.test.js

```diff
@@ -231,4 +231,65 @@ describe('capturePostgres', function() {
       });
     });
   });
+
+
+
+  describe('#passPgParams', function() {
+    var postgres, query, queryObj, sandbox, segment, subsegment;
+
+    before(function() {
+      postgres = { Client: { prototype: {
+        query: function () {},
+        host: 'database.location',
+        database: 'myTestDb',
+        connectionParameters: {
+          user: 'mcmuls',
+          host: 'database.location',
+          port: '8080',
+          database: 'myTestDb'
+        }
+      }}};
+      postgres = capturePostgres(postgres);
+
+      query = postgres.Client.prototype.query;
+      postgres = postgres.Client.prototype;
+    });
+
+    beforeEach(function() {
+      segment = new Segment('test');
+      subsegment = segment.addNewSubsegment('testSub');
+
+      queryObj = {};
+
+      postgres.__query = function(args) {
+        this._queryable = true;
+        this.queryQueue = [ null, null, queryObj ];
+        Object.assign(queryObj, args);
+
+        return queryObj;
+      };
+
+      sandbox = sinon.createSandbox();
+      sandbox.stub(AWSXRay, 'getSegment').returns(segment);
+      sandbox.stub(segment, 'addNewSubsegment').returns(subsegment);
+      sandbox.stub(AWSXRay, 'isAutomaticMode').returns(true);
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should pass down raw sql query', function() {
+      query.call(postgres, 'sql here', ['values']);
```

Comment:
Can we also assert that `addNewSubsegment` is `calledOnce` in both of these tests? Just to validate we are in fact correctly patching these queries in addition to propagating the params.

Reply:
sure, done